### PR TITLE
refactor/options

### PR DESCRIPTION
**Pull request by iikirilov**
## PR Description

group options into topical classes and add as `Mixin` to the main command

follow up from #1419

### CODE REVIEW

File: artemis/src/main/java/tech/pegasys/artemis/cli/options/DataOptions.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      paramLabel = "<FILENAME>",
+      description = "Path to output data files",
+      arity = "1")
+  private String dataPath = DEFAULT_DATA_PATH;
+
+  @CommandLine.Option(
+      names = {DATA_STORAGE_MODE_OPTION_NAME},
+      paramLabel = "<STORAGE_MODE>",
+      description =
+          "Sets the strategy for handling historical chain state.  Supported values include: 'prune', and 'archive'",
```
Comment (mbaxter) [inline artemis/src/main/java/tech/pegasys/artemis/cli/options/DataOptions.java]:
```suggestion
          "Sets the strategy for handling historical chain data.  Supported values include: 'prune', and 'archive'",
```

File: artemis/src/main/java/tech/pegasys/artemis/cli/options/DataOptions.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      arity = "1")
+  private String dataStorageMode = DEFAULT_DATA_STORAGE_MODE;
```
Comment (mbaxter) [inline artemis/src/main/java/tech/pegasys/artemis/cli/options/DataOptions.java:39]: Would be nice to [use enums](https://github.com/PegaSysEng/teku/blob/a99f47058d37d21f57567b632a9cffe9c03e3ddc/storage/src/main/java/tech/pegasys/artemis/storage/StateStorageMode.java#L18) where possible similar to how we [handle enums in besu](https://github.com/mbaxter/besu/blob/4fa78d61a3571a098ecbaa0c9980f7475ae7ca7a/besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java#L327-L332).  But its a bit off-topic, so fine to leave that for a future refactor (not necessarily completed by you).

**Review by mbaxter — APPROVED** [review]
Nice!! 💯

File: artemis/src/main/java/tech/pegasys/artemis/cli/options/DataOptions.java (hunk 2 above)
Comment (iikirilov) [reply artemis/src/main/java/tech/pegasys/artemis/cli/options/DataOptions.java:39]: that is on the list of follow up items :))

**Pull request merged**